code/modules/events/holiday/vday.dm

```diff
@@ -0,0 +1,111 @@
+
+/obj/item/weapon/valentine
+	name = "valentine"
+	desc = "A Valentine's card! Wonder what it says..."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "sc_Ace of Hearts_syndicate" // shut up
+	var/message = "A generic message of love or whatever."
+	burn_state = FLAMMABLE
+	w_class = 1
+
+/obj/item/weapon/valentine/New()
+	..()
+	message = pick("Roses are red / Violets are good / One day while Andy...",
+	               "My love for you is like the singularity. It cannot be contained.",
+	               "Will you be my lusty xenomorph maid?",
+	               "We go together like the clown and the external airlock.",
+	               "Roses are red / Liches are wizards / I love you more than a whole squad of lizards.",
+	               "Be my valentine. Law 2.",
+	               "You must be a mime, because you leave me speechless.",
+	               "I love you like Ian loves the HoP.",
+	               "You're hotter than a plasma fire in toxins.",
+	               "Are you a rogue atmos tech? Because you're taking my breath away.",
+	               "Could I have all access... to your heart?",
+	               "Call me the CMO, because I'm here to inspect your johnson.",
+	               "I'm not a changeling, but you make me extend my proboscis.",
+	               "I just can't get EI NATH of you.",
+	               "You must be a nuke op, because you make my heart explode.",
+	               "Roses are red / Chickens live on farms / Not being my Valentine / causes human harm.",
+	               "I want you more than an assistant wants insulated gloves.",
+	               "If I was a security officer, I'd brig you all shift.",
+	               "Are you the janitor? Because I think I've fallen for you.",
+	               "You're always valid to my heart.",
+	               "I'd risk the wrath of the gods to bwoink you.",
+	               "You look as beautiful now as the last time you were cloned.",
+	               "Your love is more valuable than raw plasma ore.",
+	               "Someone check the gravitational generator, because I'm only attracted to you.",
+	               "If I were the warden I'd always let you into my armory.",
+	               "The virologist is rogue, and the only cure is a kiss from you.",
+	               "Would you spend some time in my upgraded sleeper?",
+	               "You must be a silicon, because you've unbolted my heart."
+	               "Are you Nar-Sie? Because there's nar-one else I sie." )
+
+/obj/item/weapon/valentine/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/pen) || istype(W, /obj/item/toy/crayon))
+		var/recipient = stripped_input(user, "Who is receiving this valentine?", "To:", null , 20)
+		var/sender = stripped_input(user, "Who is sending this valentine?", "From:", null , 20)
+		if(recipient && sender)
+			name = "valentine - To: [recipient] From [sender]"
+	..()
+
+/obj/item/weapon/valentine/examine(mob/user)
```

Comment:
well i still think it would work better if it inherited some of this from paper
it would also help once we get the paper tgui